### PR TITLE
Feat/inheritance pin recovery dialog

### PR DESCRIPTION
**Pull request by AryanCypherock**

### CODE REVIEW

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        onPrimaryClick={() => {
+          // TODO: implement this function
+        }}
+        secondaryActionText={lang.strings.buttons.exit}
+        onSecondaryClick={() => {
+          // TODO: implement this function
+        }}
+      />
+    );
+  }
```
Comment (TejasvOnly) [inline packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx]: This should be handled in previous tab according to figma

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/index.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        $maxHeight="90vh"
+        onClose={onClose}
+      >
+        <>
+          <MilestoneAside
+            milestones={tabs
+              .filter(t => !t.dontShowOnMilestone)
+              .map(t => t.name)}
+            activeTab={currentTab}
+            heading={lang.strings.inheritanceSilverPlanPurchase.title}
```
Comment (TejasvOnly) [inline packages/cysync-core/src/dialogs/Inheritance/PinRecovery/index.tsx]: Incorrect heading

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/WalletAuth.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import { selectLanguage, useAppSelector } from '~/store';
+
+import { useInheritancePinRecoveryDialog } from '../context';
+import { Layout } from '../Layout';
+
+const checkIconComponent = <Check width={15} height={12} />;
+const throbberComponent = <Throbber size={15} strokeWidth={2} />;
+const rightArrowIcon = <ArrowRightIcon />;
+
+export const WalletAuth = () => {
```
Comment (TejasvOnly) [inline packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/WalletAuth.tsx]: Why is this component called `WalletAuth`?

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/Success.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { useInheritancePinRecoveryDialog } from '../context';
+
+export const SuccessPinRecovery = () => {
+  const lang = useAppSelector(selectLanguage);
+  const { onClose } = useInheritancePinRecoveryDialog();
+
+  const strings = lang.strings.dialogs.inheritancePinRecovery.success;
+
+  return (
+    <SuccessDialog
```
Comment (TejasvOnly) [inline packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/Success.tsx]: Need a confetti blast here

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        </Typography>
+        <Typography $fontSize={16} $textAlign="center" color="muted" mb={2}>
+          {strings.decrypted.subTitle}
+        </Typography>
+      </Container>
+      <MessageBox
+        type="success"
+        icon={<AccountIcon />}
+        text={strings.decrypted.messageBox.info}
+      />
```
Comment (TejasvOnly) [inline packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx]: This is different from figma

File: packages/ui/src/components/molecules/MessageBox.tsx [hunk 6]
```diff
@@ -11,22 +11,25 @@ import {
 } from '../atoms';
 import { BgColor, BorderColor } from '../utils';
 
-export type MessageBoxType = 'info' | 'warning' | 'danger';
+export type MessageBoxType = 'info' | 'warning' | 'danger' | 'success';
```
Comment (TejasvOnly) [inline packages/ui/src/components/molecules/MessageBox.tsx]: I think figma design has been updated and this is no longer required

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx (hunk 1 above)
Comment (TejasvOnly) [reply packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx]: Also the design is different in figma

**Review by TejasvOnly — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/WalletAuth.tsx (hunk 3 above)
Comment (AryanCypherock) [reply packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/WalletAuth.tsx]: I've changed the name from `WalletAuth` to `DecryptPin` as it describes the functionality of the component more

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx (hunk 5 above)
Comment (AryanCypherock) [reply packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx]: fixed

File: packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx (hunk 1 above)
Comment (AryanCypherock) [reply packages/cysync-core/src/dialogs/Inheritance/PinRecovery/Dialogs/DecryptedPin.tsx]: - [x] moved to previous dialog
- [x] The figma design for this component is not updated, I've to use the existing ErrorDialog component

**Review by TejasvOnly — APPROVED** [review]

**Pull request merged**